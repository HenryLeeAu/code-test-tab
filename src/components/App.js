import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchRacingList } from 'actions';
class App extends Component {
  componentDidMount() {
    this.props.fetchRacingList();
  }
  render() {
    console.log(this.props.pageStatus.sorted_data);
    const newArr = this.props.pageStatus.sorted_data.filter(item => {
      return item.meeting.raceType === this.props.pageStatus.filter;
    });
    console.log(newArr);
    return <div className="App">home</div>;
  }
}

function mapStateToProps(state) {
  return {
    pageStatus: state.pageStatus,
  };
}

export default connect(
  mapStateToProps,
  { fetchRacingList }
)(App);
