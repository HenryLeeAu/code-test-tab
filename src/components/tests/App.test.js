import React from 'react';
import App from 'components/App';
import { shallow } from 'enzyme';
let wrappered;

describe('App', () => {
  it('render UI ', () => {
    wrappered = shallow(<App />);
    expect(wrappered).toMatchSnapshot();
  });
});
