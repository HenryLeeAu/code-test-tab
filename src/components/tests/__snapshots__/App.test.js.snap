// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render UI  1`] = `
<div
  className="App"
>
  <div
    className="sidebar"
  >
    <div
      className="movieListWrap"
    />
    <div
      className="logo"
    >
      <span>
        Lucky 7
      </span>
       Hub
    </div>
    <Connect(GenreList) />
    <Connect(ProfileBox) />
  </div>
  <div
    className="detail"
  >
    <Connect(PageToolBar) />
    <Connect(PageTitle) />
    <Connect(PageSearchText) />
    <Connect(MovieList) />
  </div>
</div>
`;
